cpp/tests/io/csv_test.cpp

```diff
@@ -2244,6 +2244,27 @@ TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
   EXPECT_EQ(new_table_and_metadata.metadata.column_names[1], "1");
 }
 
+TEST_F(CsvReaderTest, EmptyColumns)
+{
+  // First column only has empty fields. second column contains only "null" literals
+  std::string csv_in{",null\n,null"};
+
+  cudf::io::csv_reader_options in_opts =
+    cudf::io::csv_reader_options::builder(cudf::io::source_info{csv_in.c_str(), csv_in.size()})
+      .names({"a", "b", "c", "d"})
+      .header(-1);
+  // More elements in `names` than in the file; additional columns are filled with nulls
+  auto result = cudf::io::read_csv(in_opts);
+
+  const auto result_table = result.tbl->view();
+  ASSERT_EQ(result_table.num_columns(), 4);
+  // All columns should contain only nulls; expect INT8 type to use as little memory as possible
+  for (auto& column : result_table) {
+    ASSERT_EQ(column.type(), data_type{type_id::INT8});
+    ASSERT_EQ(column.null_count(), 2);
```

Comment:
I was wondering what is the difference between `EXPECT_EQ` and `ASSERT_EQ` since the latter is more seen in our tests. It looks like the two are really the same according to [user guide](http://google.github.io/googletest/reference/assertions.html#EXPECT_EQ).